From f22aa341857e31d711443a1bc9dee6e2c6098430 Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Wed, 14 Feb 2018 14:09:34 +0500
Subject: [PATCH 42/42] cpu: disable IBPB

We'd like to disable IBPB for now on the AMD builds as
that tends to hit the performance quite noticeably.

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 arch/x86/kernel/cpu/bugs.c   | 5 +++--
 arch/x86/kernel/cpu/common.c | 2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/x86/kernel/cpu/bugs.c b/arch/x86/kernel/cpu/bugs.c
index 957ad44..065220b 100644
--- a/arch/x86/kernel/cpu/bugs.c
+++ b/arch/x86/kernel/cpu/bugs.c
@@ -297,11 +297,12 @@ static void __init spectre_v2_select_mitigation(void)
 		pr_info("Filling RSB on context switch\n");
 	}
 
-	/* Initialize Indirect Branch Prediction Barrier if supported */
+	/* Initialize Indirect Branch Prediction Barrier if supported 
 	if (boot_cpu_has(X86_FEATURE_IBPB)) {
 		setup_force_cpu_cap(X86_FEATURE_USE_IBPB);
 		pr_info("Enabling Indirect Branch Prediction Barrier\n");
-	}
+	} */
+	pr_info("IBPB disabled!\n");
 }
 
 #undef pr_fmt
diff --git a/arch/x86/kernel/cpu/common.c b/arch/x86/kernel/cpu/common.c
index 08e89ed..f29bbd3 100644
--- a/arch/x86/kernel/cpu/common.c
+++ b/arch/x86/kernel/cpu/common.c
@@ -732,7 +732,7 @@ static void init_speculation_control(struct cpuinfo_x86 *c)
 	 */
 	if (cpu_has(c, X86_FEATURE_SPEC_CTRL)) {
 		set_cpu_cap(c, X86_FEATURE_IBRS);
-		set_cpu_cap(c, X86_FEATURE_IBPB);
+		//set_cpu_cap(c, X86_FEATURE_IBPB);
 	}
 	if (cpu_has(c, X86_FEATURE_INTEL_STIBP))
 		set_cpu_cap(c, X86_FEATURE_STIBP);
-- 
2.7.4

